services/core-web/src/components/Forms/incidents/IncidentForm.tsx

```diff
@@ -142,794 +143,820 @@ const retrieveInitialReportDynamicValidation = (childProps) => {
   };
 };
 
-const renderInitialReport = (incidentCategoryCodeOptions, locationOptions, isEditMode) => {
```

Comment:
This is mostly just moving all of these renders inside the component so they get updated fully if the user navigates between different incidents.